File: sdks/csharp/src/Sudomimus.Token/RefreshTokenBody.cs
using System.Text.Json.Serialization;

namespace Sudomimus.Token;

/// <summary>
/// The body (payload) claims carried in a Sudomimus refresh token.
/// </summary>
public sealed record RefreshTokenBody
{
    [JsonPropertyName("iss")]
    public required string Issuer { get; init; }

    [JsonPropertyName("aud")]
    public required string Audience { get; init; }

    [JsonPropertyName("sid")]
    public required string SessionId { get; init; }

    [JsonPropertyName("jti")]
    public required string JwtId { get; init; }

    [JsonPropertyName("iat")]
    public required long IssuedAt { get; init; }

    [JsonPropertyName("exp")]
    public required long ExpiresAt { get; init; }

    [JsonPropertyName("rotationVersion")]
    public required long RotationVersion { get; init; }
}

File: sdks/java/token/src/main/java/com/sudomimus/token/RefreshTokenBody.java
package com.sudomimus.token;

import com.fasterxml.jackson.annotation.JsonProperty;

/** Body (payload) claims carried in a Sudomimus refresh token. */
public final class RefreshTokenBody {

    @JsonProperty("iss") public String issuer;
    @JsonProperty("aud") public String audience;
    @JsonProperty("sid") public String sessionId;
    @JsonProperty("jti") public String jwtId;
    @JsonProperty("iat") public Long issuedAt;
    @JsonProperty("exp") public Long expiresAt;
    @JsonProperty("rotationVersion") public Long rotationVersion;
}
